theano/compile/tests/test_function_module.py

```diff
@@ -383,6 +383,13 @@ def test_copy_delete_updates(self):
             assert cpy(1)[0] == 4
             assert cpy(1)[0] == 4
 
+        # Test if unused implicit inputs from delete_updates are ignored
+        # as intended.
+        for mode in ["FAST_RUN", "FAST_COMPILE"]:
+            ori = theano.function([x], x, mode=mode, updates={z: z * 2})
+            cpy = ori.copy(delete_updates=True)
+
```

Comment:
Ok, I pushed the test. I had to use a new variable w, with y it complained
that you can use shared variables as inputs.

Also I dit "git push" without arguments by accident, there might be two
similar commits because of that.

On Fri, Apr 22, 2016 at 10:10 AM, Frédéric Bastien <notifications@github.com
> wrote:

> In theano/compile/tests/test_function_module.py
> <https://github.com/Theano/Theano/pull/4419#discussion_r60744209>:
>
> > @@ -383,6 +383,13 @@ def test_copy_delete_updates(self):
> >              assert cpy(1)[0] == 4
> >              assert cpy(1)[0] == 4
> >
> > +        # Test if unused implicit inputs from delete_updates are ignored
> > +        # as intended.
> > +        for mode in ["FAST_RUN", "FAST_COMPILE"]:
> > +            ori = theano.function([x], x, mode=mode, updates={z: z * 2})
> > +            cpy = ori.copy(delete_updates=True)
> > +
>
> They cover different case. The current test make a shared variable input
> as non needed anymore. The new test make an explicit input not needed
> anymore. The second case make it more difficult to remove the inputs that
> are not needed, as we can't get rid of that input as it would change the
> user interface.
> … <#m_-8859374693032350296_>
> On Fri, Apr 22, 2016 at 9:43 AM, Simon Lefrancois ***@***.***> wrote: In
> theano/compile/tests/test_function_module.py <#4419 (comment)
> <https://github.com/Theano/Theano/pull/4419#discussion_r60740267>>: > @@
> -383,6 +383,13 @@ def test_copy_delete_updates(self): > assert cpy(1)[0] ==
> 4 > assert cpy(1)[0] == 4 > > + # Test if unused implicit inputs from
> delete_updates are ignored > + # as intended. > + for mode in ["FAST_RUN",
> "FAST_COMPILE"]: > + ori = theano.function([x], x, mode=mode, updates={z: z
> * 2}) > + cpy = ori.copy(delete_updates=True) > + Does it make the first
> test redundant, or do they cover separate cases? …
> <#m_-3165024369910762941_> On Fri, Apr 22, 2016 at 8:03 AM, Frédéric
> Bastien ***@***.***> wrote: In theano/compile/tests/test_function_module.py
> <#4419 <https://github.com/Theano/Theano/pull/4419> (comment) <#4419
> (comment)
> <https://github.com/Theano/Theano/pull/4419#discussion_r60728296>>>: > @@
> -383,6 +383,13 @@ def test_copy_delete_updates(self): > assert cpy(1)[0] ==
> 4 > assert cpy(1)[0] == 4 > > + # Test if unused implicit inputs from
> delete_updates are ignored > + # as intended. > + for mode in ["FAST_RUN",
> "FAST_COMPILE"]: > + ori = theano.function([x], x, mode=mode, updates={z: z
> * 2}) > + cpy = ori.copy(delete_updates=True) > + Can you add this extra
> test that make one explicit input useless ori = theano.function([x, y], x,
> mode=mode, updates={z: z + y}) cpy = ori.copy(delete_updates=True) — You
> are receiving this because you authored the thread. Reply to this email
> directly or view it on GitHub <
> https://github.com/Theano/Theano/pull/4419/files/b96e3954b09296326f768af7fc3eae85eb7b87ab..03281b5f61908a23c3cdb07f78f80607b307caa3#r60728296>
> — You are receiving this because you commented. Reply to this email
> directly or view it on GitHub <
> https://github.com/Theano/Theano/pull/4419/files/b96e3954b09296326f768af7fc3eae85eb7b87ab..03281b5f61908a23c3cdb07f78f80607b307caa3#r60740267>
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/4419/files/b96e3954b09296326f768af7fc3eae85eb7b87ab..03281b5f61908a23c3cdb07f78f80607b307caa3#r60744209>
>